src/hdlc.rs

```diff
@@ -1,17 +1,56 @@
 use alloc::borrow::Cow;
 use hdlcparse::type3::HdlcFrame;
-use nom::{branch::alt, bytes::complete::tag, sequence::tuple};
+use nom::{number::complete::u8, sequence::tuple};
 
 use crate::{DlmsDataLinkLayer, Error};
 
-fn parse_llc_header(input: &[u8]) -> Option<&[u8]> {
-  let (input, _) = tuple::<_, _, (), _>((
-    tag([0xE6]),
-    alt((tag([0xE6]), tag([0xE7]), tag([0xFF]))),
-    tag([0x00]),
-  ))(input)
-  .ok()?;
-  Some(input)
+enum Destination {
+  Unicast,
+  Broadcast,
+}
+
+enum MessageType {
+  Command,
+  Response,
+}
+
+struct LlcHeader {
+  destination: Destination,
+  message_type: MessageType,
+  quality: u8,
+}
+
+fn parse_llc_header(input: &[u8]) -> Option<(&[u8], LlcHeader)> {
+  let (input, (dest_lsap, src_lsap, quality)) = tuple::<_, _, (), _>((u8, u8, u8))(input).ok()?;
+  let destination = match dest_lsap {
+    0xE6 => Destination::Unicast,
+    0xFF => Destination::Broadcast,
+    _ => return None,
+  };
+  let message_type = match src_lsap {
+    0xE6 => MessageType::Command,
+    0xE7 => MessageType::Response,
+    _ => return None,
+  };
+
+  Some((
+    input,
+    LlcHeader {
+      destination,
+      message_type,
+      quality,
+    },
+  ))
+}
+
+fn validate_llc_header(llc: &LlcHeader) -> bool {
+  llc.quality == 0x00
+    && match llc.destination {
+      Destination::Unicast | Destination::Broadcast => true,
+    }
+    && match llc.message_type {
```

Comment:
These matches seem useless now.

Reply:
I put them there to get rid of the unused variable warnings. Should I remove them?

Reply:
I think we can remove the `validate_llc_header` function and move the `quality` check also in the parse function.

Reply:
But then the `LlcHeader` struct is useless

Reply:
Maybe just add `allow(unused)` to the header fields then. We don't really need them, but I think parsing the full header makes the code more self-documenting.